app/datarequest/add.js

```diff
@@ -106,37 +106,21 @@ class Container extends React.Component {
     }
 };
 
-function submitData(data)
+async function submitData(data)
 {
     // Disable submit button
     $("button:submit").attr("disabled", "disabled");
 
-    var tokenName = form.dataset.csrf_token_name;
-    var tokenHash = form.dataset.csrf_token_hash;
-
-    // Create form data.
-    var bodyFormData = new FormData();
-    bodyFormData.set(tokenName, tokenHash);
-    bodyFormData.set('formData', JSON.stringify(data));
-
-    // If set, append previous_request_id to POST data
-    if (typeof(previousRequestId) !== 'undefined') {
-        bodyFormData.set('previousRequestId', previousRequestId);
+    // Store
+    try {
```

Comment:
Is there a particular reason for the try/catch-block instead of the catch() function on the resulting Promise?

Reply:
No, I think using the catch() function on the resulting promise would be better. This is a just some legacy.